File: api/BeSureApi/Models/Tenant.cs
using System.ComponentModel.DataAnnotations;

namespace BeSureApi.Models
{
    public class Tenant
    {
        [Required(ErrorMessage = "validation_error_tenant_create_name_required")]
        public string Name { get; set; }
        [Required(ErrorMessage = "validation_error_tenant_create_name_in_print_required")]
        public string NameOnPrint { get; set; }
        [Required(ErrorMessage = "validation_error_tenant_create_address_required")]
        public string Address { get; set; }
    }
    public class TenantUpdate
    {
        [Required(ErrorMessage = "validation_error_tenant_create_name_required")]
        public string Name { get; set; }
        [Required(ErrorMessage = "validation_error_tenant_create_name_in_print_required")]
        public string NameOnPrint { get; set; }
        [Required(ErrorMessage = "validation_error_tenant_create_address_required")]
        public string CWHAddress { get; set; }
        [Required(ErrorMessage = "validation_error_tenant_create_address_required")]
        public string GRCAddress { get; set; }
        [Required(ErrorMessage = "validation_error_tenant_create_address_required")]
        public string HOAddress { get; set; }
        public int Id { get; set; }
        public int TenantId { get; set; }
        public int City { get; set; }
        public int Country { get; set; }
        public int State { get; set; }
        public int CWHId { get; set; }
        public int GRCId { get; set; }
        public int HDOFId { get; set; }
        public string Address { get; set; }
        public string Pincode { get; set; }
        public string PanNumber { get; set; }
    }
    public class TenantList
    {
        public int Id { get; set; }
        public string TenantCode { get; set; }
        public string TenantId { get; set; }
        public string Name { get; set; }       
        public string NameOnPrint { get; set; }
        public string IsVerified { get; set; }
        public DateTime CreatedOn { get; set; }
    }
}

File: ui/src/types/tenant.ts
import { Decoder, array, boolean, nullable, number, object, string } from "decoders";

export interface TenentForCreation {
  Name: string;
  NameOnPrint: string;
  Address: string;

}
export interface TenantData {
  IsTenantCreated: Boolean;
}

export const tenantDataDecoder: Decoder<TenantData> = object({
  IsTenantCreated: boolean
})

export interface TenantList {
  Id: number,
  TenantCode: string,
  TenantId: string,
  Name: string;
  NameOnPrint: string;
  CreatedOn: string;
  IsVerified: string;
}

export const tenantDetailsDecoder: Decoder<TenantList> = object({
  Id: number,
  TenantCode: string,
  TenantId: string,
  Name: string,
  NameOnPrint: string,
  CreatedOn: string,
  IsVerified: string
});

export interface MultipleTenantDetails {
  Tenants: TenantList[];
  TotalRows: number;
  PerPage: number;
}

export const multipleTenantDetailsDecoder: Decoder<MultipleTenantDetails> = object({
  Tenants: array(tenantDetailsDecoder),
  TotalRows: number,
  PerPage: number
});

export interface SelectedTenantDetails {
  TenantCode: string;
  Name: string;
  NameOnPrint: string;
  CreatedOn: string;
  CreatedBy: string;
  EffectiveFrom: string;
  EffectiveTo: string | null;
  CWHAddress: string;
  GRCAddress: string;
  HOAddress: string;
  PanNumber: string;
  Country: string;
  TenantState: string;
  City: string;
  Pincode: string;
  Address: string;
}
export const selectedtenantDetailDecoder: Decoder<SelectedTenantDetails> = object({
  TenantCode: string,
  Name: string,
  NameOnPrint: string,
  CreatedOn: string,
  CreatedBy: string,
  EffectiveFrom: string,
  EffectiveTo: nullable(string),
  CWHAddress: string,
  GRCAddress: string,
  HOAddress: string,
  PanNumber: string,
  Country: string,
  TenantState: string,
  City: string,
  Pincode: string,
  Address: string

});

export interface SelectedTenant {
  TenantDetails: SelectedTenantDetails[];
}

export const selectedtenantDetailsDecoder: Decoder<SelectedTenant> = object({
  TenantDetails: array(selectedtenantDetailDecoder),
});

export interface TenantUpdateDetails {
  TenantId?: number
  Name: string;
  NameOnPrint: string;
  PanNumber: string;
  CWHAddress: string;
  GRCAddress: string;
  HOAddress: string;
  Address: string;
  Pincode: string,
  City: number;
  State: number;
  Country: number;
  Id: number;
  CWHId: number,
  GRCId: number,
  HDOFId: number,
}
export const tenantUpdateDetailDecoder: Decoder<TenantUpdateDetails> = object({
  Name: string,
  NameOnPrint: string,
  PanNumber: string,
  CWHAddress: string,
  GRCAddress: string,
  HOAddress: string,
  Id: number,
  Address: string,
  City: number,
  State: number,
  Country: number,
  Pincode: string,
  CWHId: number,
  GRCId: number,
  HDOFId: number,
});

export interface TenantUpdate {
  TenantDetails: TenantUpdateDetails
}

export const tenantUpdateDetailsDecoder: Decoder<TenantUpdate> = object({
  TenantDetails: (tenantUpdateDetailDecoder),
});

export interface UpdateTenant {
  IsTenantUpdated: Boolean;
}

export const updateTenantDecoder: Decoder<UpdateTenant> = object({
  IsTenantUpdated: boolean
})

export interface Select {
  value: any,
  label: any,
  code?: any
}

export interface SelectDetails {
  Select: Select[]
}

export interface TenantSelectDetails {
  States: Select[],
  Countrys: Select[],
  Cities: Select[],
}